addons/base/models.py

```diff
@@ -38,7 +40,8 @@
 
 
 class BaseAddonSettings(ObjectIDMixin, BaseModel):
-    deleted = models.BooleanField(default=False)
+    is_deleted = models.BooleanField(default=False)
+    deleted = NonNaiveDateTimeField(null=True, blank=True)
```

Comment:
Can we just migrate `deleted` to a datetime and make `is_deleted` a property along the lines of `return self.deleted is None.` You could add a setter to it as was to retian the ergonomic quality, you can also query it as `deleted__isnull=True` for undeleted files.

Reply:
`is_deleted` will eventually be removed, so I'd be hesitant to make an entire new property for something that will be removed soon. The point of changing the name to `is_deleted` is to retain deleted data, while clearing the namespace for the new `deleted` date.

I do see your concerns, however, especially before the migration finishes and deleted is not yet populated.